metrics-operator/controllers/common/providers/provider_test.go

```diff
@@ -12,35 +12,58 @@ import (
 
 func TestFactory(t *testing.T) {
 	tests := []struct {
-		name     string
-		provider interface{}
-		err      bool
+		name         string
+		providerType string
+		provider     interface{}
+		err          bool
 	}{
 		{
-			name:     PrometheusProviderName,
-			provider: &prometheus.KeptnPrometheusProvider{},
-			err:      false,
+			name:         PrometheusProviderType,
+			providerType: "",
+			provider:     &prometheus.KeptnPrometheusProvider{},
+			err:          false,
 		},
 		{
-			name:     DynatraceProviderName,
-			provider: &dynatrace.KeptnDynatraceProvider{},
-			err:      false,
+			name:         DynatraceProviderType,
+			providerType: "",
+			provider:     &dynatrace.KeptnDynatraceProvider{},
+			err:          false,
 		},
 		{
-			name:     DataDogProviderName,
-			provider: &datadog.KeptnDataDogProvider{},
-			err:      false,
+			name:         DataDogProviderType,
+			providerType: "",
+			provider:     &datadog.KeptnDataDogProvider{},
+			err:          false,
 		},
 		{
-			name:     "invalid",
-			provider: nil,
-			err:      true,
+			name:         PrometheusProviderType,
+			providerType: PrometheusProviderType,
+			provider:     &prometheus.KeptnPrometheusProvider{},
+			err:          false,
+		},
+		{
+			name:         DynatraceProviderType,
+			providerType: DynatraceProviderType,
+			provider:     &dynatrace.KeptnDynatraceProvider{},
+			err:          false,
+		},
+		{
+			name:         DataDogProviderType,
+			providerType: DataDogProviderType,
+			provider:     &datadog.KeptnDataDogProvider{},
+			err:          false,
+		},
+		{
```

Comment:
let's add a test where name is `DynatraceProviderType` and providerType is `PrometheusProviderType`, in this case `PrometheusProviderType` should be the result

Reply:
done